contracts/MerkleProof.slb

```diff
@@ -0,0 +1,28 @@
+pragma solidity ^0.4.24;
+
+
+library MerkleProof {
+    /**
+     * @dev Verifies a Merkle proof proving the existence of a leaf in a Merkle tree
+     * @param proof Merkle proof containing sibling hashes on the branch from the leaf to the root of the Merkle tree
+     * @param root Merkle root
+     * @param position the expected position of the leaf
+     */
+    function verify(bytes32[] proof, bytes32 root, bytes32 leaf, uint256 position) internal pure returns (bool) {
```

Comment:
@peara Mine is different; you can take a look at how it is done starting by the leaves here:
https://github.com/leapdao/solEVM-enforcer/pull/28/files#diff-150ae614cbb6a572eaf82cbb80eeb215R69

Is this function here for verifying the whole path to a leaf on-chain? 
If so, I don't know if that is feasible because it could cost a lot of gas. 🤔

Reply:
@yes, it verify from leaf to root on chain. In extreme case it could be 256 caculation. It costs a lot but it required :thinking: